pkg/event/target/mqtt.go

```diff
@@ -205,14 +180,13 @@ func NewMQTTTarget(id string, args MQTTArgs) (*MQTTTarget, error) {
 		args:   args,
 		client: client,
 		store:  store,
-		reconn: false,
 	}
 
-	// Replay any previously persisted events in the store.
-	if len(target.store.ListAll()) != 0 {
-		go target.retry()
+	// Replays the events from the store.
+	eventKeyCh, getNextKeyCh := ReplayEvents(target.store, doneCh)
 
-	}
+	// Start replaying events from the store.
+	go event.SendEvents(target, eventKeyCh, getNextKeyCh, doneCh)
```

Comment:
You should not import parent in child.  Its not recommended.

Reply:
Done